18_regexp_lab/01.py

```diff
@@ -1,12 +1,27 @@
-"""
-Write a program that reads data
-from property files.
-Each line in the file can either be:
-    An empty line
-    A comment line (Start with #)
-    A property line (of the form key = value)
-
-Write a program that takes a property file name and key
-as command line arguments and prints the requested value
-"""
-
+'''
+Receives a file in format
+x1 = y1
+x2 = y2
+...
+and a key (x) and prints the value of x (y)
+Uses regex to find relevant line
+'''
+
+import sys
+import os
+import re
+
+if len(sys.argv) != 3:
+    print 'Usage: %s <file_name> <key>' % sys.argv[0]
+    sys.exit(1)
+
+if not os.path.exists(sys.argv[1]):
+    print 'File: %s does not exist' % sys.argv[1]
+    sys.exit(1)
+
+with open(sys.argv[1], 'r') as f:
+    for line in f:
+        result = re.search(r'%s[ ]*=[ ]*(\d*)' % sys.argv[2], line)

```

Comment:
Using external input as regexp is considered bad practice: The input may contain characters that have special meaning in regexp, such as `.`, `^`, `$`, etc. Better to read all values to a dictionary and search that.